apps/frontend/src/components/ui/servers/ServerStats.vue

```diff
@@ -28,31 +30,41 @@
         <div class="absolute -left-8 -top-4 h-28 w-56 rounded-full bg-bg-raised blur-lg" />
       </div>
 
-      <component :is="metric.icon" class="absolute right-10 top-10 z-10" />
-      <ClientOnly>
-        <VueApexCharts
-          v-if="metric.showGraph"
-          type="area"
-          height="142"
-          :options="getChartOptions(metric.warning)"
-          :series="[{ name: metric.title, data: metric.data }]"
-          class="chart absolute bottom-0 left-0 right-0 w-full opacity-0"
-        />
-      </ClientOnly>
+      <component
+        :is="metric.icon"
+        class="absolute right-10 top-10 z-10 size-8"
+        style="width: 2rem; height: 2rem"
+      />
+
+      <div class="chart-space absolute bottom-0 left-0 right-0">
+        <ClientOnly>
+          <VueApexCharts
+            v-if="metric.showGraph && !loading"
+            type="area"
+            height="142"
+            :options="getChartOptions(metric.warning, index)"
+            :series="[{ name: metric.title, data: metric.data }]"
+            class="chart opacity-0"
+            :class="{ 'chart-loaded': chartsReady.has(index) }"
```

Comment:
I assume you made the chart-loaded class because you can't "override" tailwind classes due to them having the same precedence, but you can do this to avoid that and it makes it more clear at a glance what styles are applied based on the condition

```suggestion
            :class="chartsReady.has(index) ? 'opacity-100' : 'opacity-0'"
```